internal/rbd/controllerserver.go

```diff
@@ -444,8 +444,17 @@ func (cs *ControllerServer) createVolumeFromSnapshot(ctx context.Context, cr *ut
 
 	// update parent name(rbd image name in snapshot)
 	rbdSnap.RbdImageName = rbdSnap.RbdSnapName
+	parentVol := generateVolFromSnap(rbdSnap)
+	// as we are operating on single cluster reuse the connection
+	parentVol.conn = rbdVol.conn
```

Comment:
Connections are reference counted, you can not use it like this. Just use `Connect()` with the same parameters and it should work as intended.